### DISCUSSION TITLE
Todo list

### DISCUSSION THREAD

**Discussion by alaingdl** [gnudatalanguage/gdl#1374] [General]
I don't know where to discuss on that topic except here.

Could we set up a list of open issues, problems, trying to classified the priorities ?
What I see today : 

*  _build_gdl.sh_ is not working on some configurations (u22.04, osx10.13.6 ...) This is not a detail, in labs. computers are not updated to last versions ...  I do provide support for such versions.

* how to check that the outputs of image formats (png, svg, jpeg, tiff) are OK for external tools ? Currently it is broken in the Git version but we did not realized quickly that those important functionalities were broken. I don't see how to test that in CI ...

* missing pro/func + keyword
  ** family of functions : [EIGENVEC](https://www.l3harrisgeospatial.com/docs/eigenvec.html), [ELMHES](https://www.l3harrisgeospatial.com/docs/elmhes.html), [LA_HQR](https://www.l3harrisgeospatial.com/docs/la_hqr.html#L_826528365_959357), [TRIQL](https://www.l3harrisgeospatial.com/docs/triql.html#T_809226861_52198), [TRIRED](https://www.l3harrisgeospatial.com/docs/trired.html#T_809226861_52262), EIGENQL, LA_EIGENVEC
 ** somethings else ?

* revisiting important code that are benchmarked as slow since IDL 8.8 was realized (e.g. INVERT())
* re-generating a matrix of http://aramis.obspm.fr/~coulais/IDL_et_GDL/Matrice_IDLvsGDL_missing.html 
* clarifying the status of HDF, HDF5, netcdf formats (was is working of not, what is blocking, critical, buggy ...)
* extending the test suite
* extending the benckmark suite

**Comment by GillesDuvert** [comment]
I suppose all LA_XXXXX routines are interfaces for LAPACK, itself on [GitHub](https://github.com/Reference-LAPACK/lapack)?
So this would be "just" writing an interface to a shared lib (?)

Concerning the output graphic formats (png etc) ideally we could test the identity of outputs between ISL and GDL on a complete serie of examples (much as was started for TIFF). Not necessary part of GDL (same as benchmarking) as this is not part of the main distro.

It is not completely obvious that IDL does the best treatment when reading or writing images (what with the 16bit images) and GDL 'just' uses what ImageMagick provides --- something more recent and probably better than IDL in this case, but of course we seek compatiblity first.
